### PR TITLE
pull with container execution credentials

### PR DESCRIPTION
**Pull request by Realmonia**
### Summary

support pulling image from ECR with container execution credentials

### Testing

make test

New tests cover the changes: 
yes

### Description for the changelog

support pulling image from ECR with container execution credentials by using containerExecutionRoleCredentials by default and back up with taskExecutionRoleCredentials

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: agent/engine/docker_task_engine.go [hunk 1]
```diff
@@ -950,18 +950,18 @@ func (engine *DockerTaskEngine) pullAndUpdateContainerReference(task *apitask.Ta
...
+				task.Arn, container.Image, container.Name, credentialsType)
 			return dockerapi.DockerContainerMetadata{
 				Error: dockerapi.CannotPullECRContainerError{
 					FromError: errors.New("engine ecr credentials: not found"),
 				},
 			}
 		}
 
-		iamCredentials := executionCredentials.GetIAMRoleCredentials()
+		seelog.Infof("Set RegistryAuthCredentials with %sExecutionCredentials", credentialsType)
```
Comment (sharanyad) [inline agent/engine/docker_task_engine.go:964]: can we also log task arn and container name for more context here?

File: agent/engine/docker_task_engine.go [hunk 2]
```diff
@@ -1023,6 +1023,16 @@ func (engine *DockerTaskEngine) pullAndUpdateContainerReference(task *apitask.Ta
 	return metadata
 }
 
+func (engine *DockerTaskEngine) getPullImageIamCredentials(task *apitask.Task, container *apicontainer.Container) (credentials.IAMRoleCredentials, bool, string) {
```
Comment (sharanyad) [inline agent/engine/docker_task_engine.go:1026]: nit: can we rename the method to `getPullImageIAMCredentials` ?

Comment (Realmonia) [reply agent/engine/docker_task_engine.go:1026]: sure

File: agent/engine/docker_task_engine.go (hunk 1 above)
Comment (Realmonia) [reply agent/engine/docker_task_engine.go:964]: will do!

**Review by yinyic — APPROVED** [review]

**Review by sharanyad — APPROVED** [review]
:shipit:

**Pull request merged**